src/main/java/carpet/api/settings/RuleHelper.java

```diff
@@ -0,0 +1,112 @@
+        return false;
+    }
+    
+    /**
+     * <p>Converts a rule's value into its similar {@link String} representation.</p>
+     * 
+     * <p>If the value is an {@link Enum}, this method returns the name of the enum constant lowercased, else
+     * it returns the result of running {@link #toString()} on the value.</p>
+     * @param value A rule's value
+     * @return A {@link String} representation of the given value
+     */
+    public static String toRuleString(Object value) {
+        if (value instanceof Enum) return ((Enum<?>) value).name().toLowerCase(Locale.ROOT);
+        return value.toString();
+    }
+    
+    /**
+     * <p>Checks whether the given {@code CarpetRule rule} is in its default value</p>
+     * 
+     * @param rule The rule to check
+     * @return {@code true} if the rule's default value equals its current value
+     */
+    public static boolean isInDefaultValue(CarpetRule<?> rule) {
+        return rule.defaultValue().equals(rule.value());
+    }
+    
+    /**
+     * <p>Resets the given {@link CarpetRule rule} to its default value, and notifies the given source, if provided.</p>
+     * @param rule The {@link CarpetRule} to reset to its default value
+     * @param source A {@link ServerCommandSource} to notify about this change, or {@code null}
+     * 
+     * @param <T> The type of the {@link CarpetRule}
+     */
+    public static <T> void resetToDefault(CarpetRule<T> rule, CommandSourceStack source) {
+        try {
+            rule.set(source, rule.defaultValue());
+        } catch (InvalidRuleValueException e) {
+            throw new IllegalStateException("Rule couldn't be set to default value!", e);
+        }
+    }
+    
+    // Translations
+    // The methods from this point are not stable API yet and may change or be removed in binary incompatible ways later
+    
+    /**
+     * @param rule The {@link CarpetRule} to get the translated name of
+     * @return A {@link String} being the translated {@link CarpetRule#name() name} of the given rule, the current language.
+     * 
+     * @apiNote This method isn't stable API yet and may change or be removed in binary incompatible ways in later Carpet versions
+     */
+    @ApiStatus.Experimental
+    public static String translatedName(CarpetRule<?> rule) {
+        String key = String.format(TranslationKeys.RULE_NAME_PATTERN, rule.settingsManager().identifier(), rule.name());
+        return Translations.hasTranslation(key) ? Translations.tr(key) + String.format(" (%s)", rule.name()): rule.name();
+    }
+    
+    /**
+     * @param rule The {@link CarpetRule} to get the translated description of
+     * @return A {@link String} being the translated description of this rule, in Carpet's configured language.
+     * 
+     * @apiNote This method isn't stable API yet and may change or be removed in binary incompatible ways in later Carpet versions
+     */
+    @ApiStatus.Experimental
```

Comment:
I know its jetbrains and feels pretty standard, but log4j also felt a staple at the time and now we need to suffer due to autocomplete decision most likely. That annotation is probably only for information purposes, so maybe not really needed? I wish java had some standard common annotations for these purposes.

Reply:
Not really, I'll remove them.

Didn't think it was bad given it's only kept up to class data and discarded at runtime, Java ignores missing annotations if they were to be removed, and loom/mc/some mc dep adds it to the compile classpath anyway. And I think with it some (jetbrains) IDEs provide warnings about them when being referenced, which is nice.

In vanilla Java this would be `@Deprecated` iinw ("_discouraged from using_"), but I thought it didn't really express what it meant given it's normally used as "this will be removed".